lib/compilers/analysis-tool.ts

```diff
@@ -30,20 +31,25 @@ export class AnalysisTool extends BaseCompiler {
         return 'analysis-tool';
     }
 
-    constructor(info, env) {
+    constructor(info: CompilerInfo & Record<string, any>, env) {
         // Default is to disable all "cosmetic" filters
         if (!info.disabledFilters) info.disabledFilters = ['labels', 'directives', 'commentOnly', 'trim'];
         super(info, env);
     }
 
-    getDefaultFilters() {
+    override getDefaultFilters() {
         // Disable everything but intel syntax
         return {
             intel: true,
             commentOnly: false,
             directives: false,
             labels: false,
             optOutput: false,
+            binary: false,
+            execute: false,
+            demangle: false,
+            libraryCode: false,
+            trim: false,
```

Comment:
Are these the right defaults?

Reply:
I think we need: `demangle: true` but everything else looks good to me. @partouf is that the right default for `libraryCode` - I think so.

Reply:
both demangle and libraryCode and are defaulted to true in the UI, but so are directives and commentOnly, I think we leave the defaults to the UI and people using the API should do everything manually. Because then they'll know what might be filtered out - instead of having everything filtered out by default and then they don't know what they're might be missing.

Reply:
I don't really know what this is doing, but it's probably missing binaryObject if the goal is to init all possible filters.

Reply:
please, do add it

Reply:
`BaseCompiler.getDefaultFilters` is implemented as https://github.com/compiler-explorer/compiler-explorer/blob/784393bbc78342197ac26ad8e9b236f55bcc74bb/lib/base-compiler.ts#L2755-L2768

It looks like the return type should be `ParseFiltersAndOutputOptions` so I will update it

Reply:
@dkm What should the default for `binaryObject` be?

Reply:
I am pretty sure `false`. I think we could "just" try that and fix it up (in order to get this merged)